File: tests/CoreTests/ExtensionMethods/ExtensionMethods.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SharpKit.JavaScript;

namespace CoreTests
{
    [JsType(JsMode.Prototype)]
    static class ExtensionMethods
    {
        public static void ExtensionMethod(this Test x)
        {
        }
        static void foo()
        {
            var xx = new Test();
            Action action = xx.ExtensionMethod;
            Action action2 = xx.InstanceMethod;

            List<Test> list = new List<Test>();
            list.ForEach(ExtensionMethod);
            xx.ExtensionMethod();
            ExtensionMethod(xx);

        }
    }

    class Test
    {
        public void InstanceMethod()
        {
        }
    }

}

File: trunk/tests/CoreTests/ExtensionMethods/ExtensionMethods.js
;;
if (typeof(CoreTests) == "undefined")
    var CoreTests = {};
CoreTests.ExtensionMethods = function (){
};
CoreTests.ExtensionMethods.ExtensionMethod = function (x){
};
CoreTests.ExtensionMethods.foo = function (){
    var xx = new CoreTests.Test.ctor();
    var action = $CreateExtensionDelegate(xx, CoreTests.ExtensionMethods.ExtensionMethod);
    var action2 = $CreateDelegate(xx, xx.InstanceMethod);
    var list = new System.Collections.Generic.List$1.ctor(CoreTests.Test.ctor);
    list.ForEach(CoreTests.ExtensionMethods.ExtensionMethod);
    CoreTests.ExtensionMethods.ExtensionMethod(xx);
    CoreTests.ExtensionMethods.ExtensionMethod(xx);
};

